Lesson23/Feeling_lucky.py

```diff
@@ -0,0 +1,240 @@
+#step of using it to improve our search results. For this question, you will use
+#the page rankings to produce the best output for a given query.
+
+#Define a procedure, lucky_search, that takes as input an index, a ranks
+#dictionary (the result of compute_ranks), and a keyword, and returns the one
+#URL most likely to be the best site for that keyword. If the keyword does not
+#appear in the index, lucky_search should return None.
+
+def lucky_search(index, ranks, keyword):
+    if lookup(index,keyword):
+        best = 0
+        lucky = ""
+        for e in lookup(index,keyword):
+            if ranks[e] > best:
+                best = ranks[e]
+                lucky = e
+        return lucky
+            
+
+
+cache = {
+   'http://udacity.com/cs101x/urank/index.html': """<html>
+<body>
+<h1>Dave's Cooking Algorithms</h1>
+<p>
+Here are my favorite recipies:
+<ul>
+<li> <a href="http://udacity.com/cs101x/urank/hummus.html">Hummus Recipe</a>
+<li> <a href="http://udacity.com/cs101x/urank/arsenic.html">World's Best Hummus</a>
+<li> <a href="http://udacity.com/cs101x/urank/kathleen.html">Kathleen's Hummus Recipe</a>
+</ul>
+
+For more expert opinions, check out the
+<a href="http://udacity.com/cs101x/urank/nickel.html">Nickel Chef</a>
+and <a href="http://udacity.com/cs101x/urank/zinc.html">Zinc Chef</a>.
+</body>
+</html>
+
+
+
+
+
+
+""",
+   'http://udacity.com/cs101x/urank/zinc.html': """<html>
+<body>
+<h1>The Zinc Chef</h1>
+<p>
+I learned everything I know from
+<a href="http://udacity.com/cs101x/urank/nickel.html">the Nickel Chef</a>.
+</p>
+<p>
+For great hummus, try
+<a href="http://udacity.com/cs101x/urank/arsenic.html">this recipe</a>.
+
+</body>
+</html>
+
+
+
+
+
+
+""",
+   'http://udacity.com/cs101x/urank/nickel.html': """<html>
+<body>
+<h1>The Nickel Chef</h1>
+<p>
+This is the
+<a href="http://udacity.com/cs101x/urank/kathleen.html">
+best Hummus recipe!
+</a>
+
+</body>
+</html>
+
+
+
+
+
+
+""",
+   'http://udacity.com/cs101x/urank/kathleen.html': """<html>
+<body>
+<h1>
+Kathleen's Hummus Recipe
+</h1>
+<p>
+
+<ol>
+<li> Open a can of garbonzo beans.
+<li> Crush them in a blender.
+<li> Add 3 tablesppons of tahini sauce.
+<li> Squeeze in one lemon.
+<li> Add salt, pepper, and buttercream frosting to taste.
+</ol>
+
+</body>
+</html>
+
+""",
+   'http://udacity.com/cs101x/urank/arsenic.html': """<html>
+<body>
+<h1>
+The Arsenic Chef's World Famous Hummus Recipe
+</h1>
+<p>
+
+<ol>
+<li> Kidnap the <a href="http://udacity.com/cs101x/urank/nickel.html">Nickel Chef</a>.
+<li> Force her to make hummus for you.
+</ol>
+
+</body>
+</html>
+
+""",
+   'http://udacity.com/cs101x/urank/hummus.html': """<html>
+<body>
+<h1>
+Hummus Recipe
+</h1>
+<p>
+
+<ol>
+<li> Go to the store and buy a container of hummus.
+<li> Open it.
+</ol>
+
+</body>
+</html>
+
+
+
+
+""",
+}
+
+def get_page(url):
+    if url in cache:
+        return cache[url]
+    return ""
```

Comment:
I would prefer to return None here as None means page doesn't exist, "" means page exists, but content is empty.

Reply:
ok